src/fsharp/FSComp.txt

```diff
@@ -1471,13 +1471,13 @@ keywordDescriptionUntypedQuotation,"Delimits a untyped code quotation."
 3216,itemNotFoundDuringDynamicCodeGen,"%s '%s' not found in assembly '%s'. A possible cause may be a version incompatibility. You may need to explicitly reference the correct version of this assembly to allow all referenced components to use the correct version."
 3216,itemNotFoundInTypeDuringDynamicCodeGen,"%s '%s' not found in type '%s' from assembly '%s'. A possible cause may be a version incompatibility. You may need to explicitly reference the correct version of this assembly to allow all referenced components to use the correct version."
 descriptionWordIs,"is"
-notAFunction,"This value is not a function and cannot be applied."
-notAFunctionButMaybeIndexerWithName,"This value is not a function and cannot be applied. Did you intend to access the indexer via '%s.[index]'?"
-notAFunctionButMaybeIndexer,"This expression is not a function and cannot be applied. Did you intend to access the indexer via 'expr.[index]'?"
-notAFunctionButMaybeIndexerWithName2,"This value is not a function and cannot be applied. Did you intend to access the indexer via '%s[index]'?"
-notAFunctionButMaybeIndexer2,"This expression is not a function and cannot be applied. Did you intend to access the indexer via 'expr[index]'?"
+notAFunction,"This is a value of type '%s', and not a function. It doesn't take any arguments, but you're trying to pass a value as an argument to it. Consider removing the extra argument."
```

Comment:
I think printing the type first will lead to confusion for large and complex types.  In each case, I think it's better to at least say something before showing the type. e.g. Move the type as follows:

`This value is not a function. it has type '%s'.`

That said, in very many cases it feels like it would be better to summarize the type more succinctly than printing it.  For example

    This value is not a function, it is a tuple.

is better than

    This value is not a function, it has type `int * int * int * int * int * int * int * int * int * int * int * int * int * int * int`.

Or perhaps:

    This value is not a function, it is a tuple, it has type `int * int * int * int * int * int * int * int * int * int * int * int * int * int * int`.

Do you think that could be adjusted for tuples, delegates, anon records?

Reply:
Makes sense to reorder the error message
About display `tuple` instead of `int * int`, I think it's possible by doing something like

```
match Ty with
    | TType_tuple(_, _) -> "tuple"
```

But I don't know if is the best approach

Reply:
You can use `isAnyTupleTy`, `isAnonRecdTy`, `isTyparTy` - these take into account type abbreviations and solved type equations